hal/targets/hal/TARGET_ARM_SSG/TARGET_BEETLE/gpio_irq_api.c

```diff
@@ -0,0 +1,244 @@
+    if (ch_bit <16){
+        if (((CMSDK_GPIO_0->INTSTATUS) == 0) || (channel_ids[channel] == 0)
+                || ((CMSDK_GPIO_0->INTTYPESET) == 0)) return;
+
+        if ((CMSDK_GPIO_0->INTTYPESET & ch_bit)
+                && (CMSDK_GPIO_0->INTPOLSET & ch_bit)) {
+            irq_handler(channel_ids[channel], IRQ_RISE);
+            CMSDK_GPIO_0->INTPOLSET = ch_bit;
+        }
+        if ((CMSDK_GPIO_0->INTTYPESET & ch_bit)
+                && ~(CMSDK_GPIO_0->INTPOLSET & ch_bit)) {
+            irq_handler(channel_ids[channel], IRQ_FALL);
+        }
+        CMSDK_GPIO_0->INTCLEAR = ch_bit;
+    }
+
+    if (ch_bit>=16) {
+        if (((CMSDK_GPIO_1->INTSTATUS) == 0) || (channel_ids[channel] == 0)
+                || ((CMSDK_GPIO_1->INTTYPESET) == 0)) return;
+
+        if ((CMSDK_GPIO_1->INTTYPESET & ch_bit)
+                && (CMSDK_GPIO_1->INTPOLSET & ch_bit)) {
+            irq_handler(channel_ids[channel], IRQ_RISE);
+            CMSDK_GPIO_1->INTPOLSET = ch_bit;
+        }
+        if ((CMSDK_GPIO_1->INTTYPESET & ch_bit)
+                && ~(CMSDK_GPIO_1->INTPOLSET & ch_bit)) {
+            irq_handler(channel_ids[channel], IRQ_FALL);
+        }
+        CMSDK_GPIO_1->INTCLEAR = ch_bit;
+    }
+}
+
+void gpio0_irq0(void) {handle_interrupt_in(0);}
+void gpio0_irq1(void) {handle_interrupt_in(1);}
+void gpio0_irq2(void) {handle_interrupt_in(2);}
+void gpio0_irq3(void) {handle_interrupt_in(3);}
+void gpio0_irq4(void) {handle_interrupt_in(4);}
+void gpio0_irq5(void) {handle_interrupt_in(5);}
+void gpio0_irq6(void) {handle_interrupt_in(6);}
+void gpio0_irq7(void) {handle_interrupt_in(7);}
+void gpio0_irq8(void) {handle_interrupt_in(8);}
+void gpio0_irq9(void) {handle_interrupt_in(9);}
+void gpio0_irq10(void) {handle_interrupt_in(10);}
+void gpio0_irq11(void) {handle_interrupt_in(11);}
+void gpio0_irq12(void) {handle_interrupt_in(12);}
+void gpio0_irq13(void) {handle_interrupt_in(13);}
+void gpio0_irq14(void) {handle_interrupt_in(14);}
+void gpio0_irq15(void) {handle_interrupt_in(15);}
+void gpio1_irq0(void) {handle_interrupt_in(16);}
+void gpio1_irq1(void) {handle_interrupt_in(17);}
+void gpio1_irq2(void) {handle_interrupt_in(18);}
+void gpio1_irq3(void) {handle_interrupt_in(19);}
+void gpio1_irq4(void) {handle_interrupt_in(20);}
+void gpio1_irq5(void) {handle_interrupt_in(21);}
+void gpio1_irq6(void) {handle_interrupt_in(22);}
+void gpio1_irq7(void) {handle_interrupt_in(23);}
+void gpio1_irq8(void) {handle_interrupt_in(24);}
+void gpio1_irq9(void) {handle_interrupt_in(25);}
+void gpio1_irq10(void) {handle_interrupt_in(26);}
+void gpio1_irq11(void) {handle_interrupt_in(27);}
+void gpio1_irq12(void) {handle_interrupt_in(28);}
+void gpio1_irq13(void) {handle_interrupt_in(29);}
+void gpio1_irq14(void) {handle_interrupt_in(30);}
+void gpio1_irq15(void) {handle_interrupt_in(31);}
+
+int gpio_irq_init(gpio_irq_t *obj, PinName pin,
+        gpio_irq_handler handler, uint32_t id) {
+    if (pin == NC) {return -1;}
```

Comment:
Fix the formatting please :-) https://developer.mbed.org/teams/SDK-Development/wiki/mbed-sdk-coding-style